File: packages/sdk-csharp/Clients/FlagsClient.cs
using Settler.Sdk.Models;

namespace Settler.Sdk;

/// <summary>Client for feature flags operations.</summary>
public sealed class FlagsClient
{
    private readonly SettlerClient _client;

    internal FlagsClient(SettlerClient client) => _client = client;

    /// <summary>Evaluate a feature flag.</summary>
    public async Task<DataResponse<FlagEvaluation>> EvaluateAsync(
        string flagKey, Dictionary<string, object>? context = null,
        CancellationToken cancellationToken = default)
    {
        var body = new Dictionary<string, object>
        {
            ["flagKey"] = flagKey,
            ["context"] = context ?? new Dictionary<string, object>(),
        };

        return await _client.RequestAsync<DataResponse<FlagEvaluation>>(
            HttpMethod.Post, "flags/evaluate", body: body, cancellationToken: cancellationToken);
    }
}

File: packages/sdk-java/src/main/java/io/settler/clients/FlagsClient.java
package io.settler.clients;

import com.fasterxml.jackson.databind.JsonNode;
import io.settler.exceptions.SettlerException;
import io.settler.http.HttpExecutor;

import java.util.HashMap;
import java.util.Map;

/**
 * Client for evaluating feature flags.
 */
public class FlagsClient {
    private final HttpExecutor executor;

    public FlagsClient(HttpExecutor executor) {
        this.executor = executor;
    }

    /**
     * Evaluates a feature flag for a given context.
     *
     * @param flagKey the key of the flag to evaluate
     * @param context the evaluation context (e.g., userId, email, etc.)
     * @param defaultValue the fallback value if evaluation fails
     * @return a JSON object containing the evaluation result
     * @throws SettlerException if the API request fails
     */
    public JsonNode evaluate(String flagKey, Map<String, Object> context, Object defaultValue) throws SettlerException {
        Map<String, Object> body = new HashMap<>();
        body.put("flagKey", flagKey);
        body.put("context", context == null ? new HashMap<>() : context);
        if (defaultValue != null) {
            body.put("defaultValue", defaultValue);
        }

        return executor.post("/v1/feature-flags/evaluate", body);
    }
}
